api/argoproj/v1alpha1/appproject.go

```diff
@@ -0,0 +1,54 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AppProject provides a logical grouping of applications, providing controls for:
+// * who can access these applications (roles, OIDC group claims bindings)
+// * and what they can do (RBAC policies)
+// +kubebuilder:object:root=true
+type AppProject struct {
```

Comment:
Maybe document in PR description, code comment or in this comment why the type are duplicated instead of importing the types from ArgoCD